src/PerfCounters_x86.h

```diff
@@ -306,10 +306,38 @@ static void check_for_zen_speclockmap() {
   }
 }
 
+static void check_for_freeze_on_smi() {
+  ScopedFd fd = ScopedFd("/sys/devices/cpu/freeze_on_smi", O_RDONLY);
+  if (!fd.is_open()) {
+    LOG(debug) << "/sys/devices/cpu/freeze_on_smi not present";
+    return;
+  }
+
+  char freeze_on_smi = 0;
+  read(fd, &freeze_on_smi, 1);
+  if (freeze_on_smi == 0) {
```

Comment:
`<= 0`

Reply:
Er, actually I meant you should check the result of the `read()`.

Reply:
Does it matter? if the read fails `freeze_on_smi` will have its original value no?